patches/net/minecraft/data/models/blockstates/VariantProperties.java.patch

```diff
@@ -0,0 +1,13 @@
+--- a/net/minecraft/data/models/blockstates/VariantProperties.java
++++ b/net/minecraft/data/models/blockstates/VariantProperties.java
+@@ -10,6 +_,10 @@
+     public static final VariantProperty<Boolean> UV_LOCK = new VariantProperty<>("uvlock", JsonPrimitive::new);
+     public static final VariantProperty<Integer> WEIGHT = new VariantProperty<>("weight", JsonPrimitive::new);
+ 
++    // Neo: Allow passing model builders into block state variants
++    // Not to be used in conjunction with 'MODEL'
++    public static final VariantProperty<net.neoforged.neoforge.client.model.generators.BlockModelBuilder> MODEL_BUILDER = new VariantProperty<>("model", model -> new JsonPrimitive(model.getLocation().toString()));
```

Comment:
I'm not sure this is particularly useful. The variant stuff is already pretty verbose, so it's unlikely someone would nest an entire model builder chain in that code, and pulling the location from a builder to pass to the vanilla `VariantProperties.MODEL` is trivial